tests_pytest/_fw_tests_common_base/base_graph_with_fusion_metadata_test.py

```diff
@@ -0,0 +1,157 @@
+from model_compression_toolkit.core.graph_prep_runner import graph_preparation_runner
+
+
+class BaseGraphWithFusingMetadataTest(abc.ABC):
+
+    fw_impl: Any
+    fw_info: Any
+    attach_to_fw_func: Callable
+
+    def _data_gen(self):
+        raise NotImplementedError()
+    def _get_model(self):
+        raise NotImplementedError()
+
+    @pytest.fixture
+    def minimal_tpc_with_fusing(self, default_quant_cfg_options):
+        """
+        Creates a minimal Target Platform Capabilities (TPC) configuration that supports specific fusions.
+        This fixture defines allowed operations and fusing patterns for testing.
+        """
+        return schema.TargetPlatformCapabilities(
+            default_qco=default_quant_cfg_options,
+            tpc_platform_type='test',
+            operator_set=[schema.OperatorsSet(name=schema.OperatorSetNames.CONV),
+                          schema.OperatorsSet(name=schema.OperatorSetNames.RELU),
+                          schema.OperatorsSet(name=schema.OperatorSetNames.FULLY_CONNECTED),
+                          schema.OperatorsSet(name=schema.OperatorSetNames.SOFTMAX)],
+            fusing_patterns=[schema.Fusing(operator_groups=(schema.OperatorsSet(name=schema.OperatorSetNames.CONV),
+                                                            schema.OperatorsSet(name=schema.OperatorSetNames.RELU))),
+                             schema.Fusing(
+                                 operator_groups=(schema.OperatorsSet(name=schema.OperatorSetNames.FULLY_CONNECTED),
+                                                  schema.OperatorsSet(name=schema.OperatorSetNames.SOFTMAX)))]
+        )
+
+    @pytest.fixture
+    def graph_with_fusion_metadata(self, minimal_tpc_with_fusing):
+        """
+        Creates a graph with fusing metadata based on a generated model and a predefined configuration.
+        Ensures all required components (framework implementation, framework info, etc.) are present.
+        """
+        assert self._data_gen is not None
+        assert self.fw_impl is not None
+        assert self.fw_info is not None
+        assert self.attach_to_fw_func is not None
+
+        self.fqc = self.attach_to_fw_func(minimal_tpc_with_fusing)
+
+        graph_with_fusion_metadata = graph_preparation_runner(self._get_model(),
+                                                              self._data_gen,
+                                                              QuantizationConfig(),
+                                                              fw_info=self.fw_info,
+                                                              fw_impl=self.fw_impl,
+                                                              fqc=self.fqc,
+                                                              mixed_precision_enable=False,
+                                                              running_gptq=False)
+        return graph_with_fusion_metadata
```

Comment:
maybe add verification that the metada exists after running `graph_preparation_runner` and that it is valid or as expected?